src/Raven.Server/Documents/Patch/V8/TaskCustomBinder.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Threading.Tasks;
+using V8.Net;
+
+
+namespace Raven.Server.Documents.Patch.V8
+{
+
+    public class TaskCustomBinder : ObjectBinderEx<Task>
+    {
+        public static InternalHandle CreateObjectBinder(V8EngineEx engine, Task oi, bool keepAlive = false) 
+        {
+            return engine.CreateObjectBinder<TaskCustomBinder>(oi, engine.TypeBinderTask, keepAlive: keepAlive);
+        }
+
+        public static InternalHandle GetRunningTaskResult(V8Engine engine, Task task)
+        {
+            var value = $"{{Ignoring Task.Result as task's status is {task.Status.ToString()}}}.";
```

Comment:
What is this used for?

Reply:
in JsBlittableBridgeV8.GetBoundObjectProperties when we create object binder:
using (InternalHandle jsObj = _engine.CreateObjectBinder(obj, keepAlive: false))
it creates TaskCustomBinder object associated with Task object

in Jint version we have similar things in JsBlittableBridgeJint.GetObjectProperties
but in V8 version non-completed task processing logic is centralised inside TaskCustomBinder, while in Jint it is in GetObjectProperties and so it is more complex